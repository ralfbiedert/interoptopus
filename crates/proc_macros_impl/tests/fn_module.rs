use quote::quote;
use syn::{parse_quote, ItemFn};

mod util;

#[test]
fn module_explicit() {
    let item: ItemFn = parse_quote! {
        #[ffi_function(module = "foo")]
        fn foo() {}
    };

    insta::assert_snapshot!(expand_fn!(item));
}

#[test]
fn module_common() {
    let item: ItemFn = parse_quote! {
        #[ffi_function(module = common)]
        fn foo() {}
    };

    insta::assert_snapshot!(expand_fn!(item));
}
