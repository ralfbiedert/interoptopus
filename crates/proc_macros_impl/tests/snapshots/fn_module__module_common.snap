---
source: crates/proc_macros_impl/tests/fn_module.rs
expression: expand_fn!(item)
---
#[unsafe(export_name = "foo")]
extern "C" fn foo() {}
#[allow(non_camel_case_types)]
struct foo {}
impl ::interoptopus::lang::function::FunctionInfo for foo {
    fn id() -> ::interoptopus::inventory::FunctionId {
        ::interoptopus::inventory::FunctionId::from_id(::interoptopus::id!(foo))
    }
    fn signature() -> ::interoptopus::lang::function::Signature {
        ::interoptopus::lang::function::Signature {
            arguments: vec![],
            rval: <() as ::interoptopus::lang::types::TypeInfo>::id(),
        }
    }
    fn function() -> ::interoptopus::lang::function::Function {
        ::interoptopus::lang::function::Function {
            name: "foo".to_string(),
            visibility: ::interoptopus::lang::meta::Visibility::Private,
            docs: ::interoptopus::lang::meta::Docs::from_lines(vec![]),
            emission: ::interoptopus::lang::meta::Emission::Common,
            signature: Self::signature(),
        }
    }
    fn register(inventory: &mut ::interoptopus::inventory::Inventory) {
        <() as ::interoptopus::lang::types::TypeInfo>::register(inventory);
        inventory.register_function(Self::id(), Self::function());
    }
}
